[#416] Remove Client Mode from the HTTP API (main) (#417)
irods/irods_client_http_api: API.md

### Scheme: OpenID Connect (OIDC)

> For authenticating with OpenID Connect, there is one method that is supported by the HTTP API. This is running the HTTP API as an OAuth Protected Resource.

Reviewer: Let's take another swing at rephrasing this. I feel this can be said in one sentence.
```suggestion
For authenticating with OpenID Connect, the HTTP API can be run as an OAuth Protected Resource.
```
By no means do you have to use what I suggested, but let's see if we can make the statement more succinct.

Reviewer 2: yes please

### Scheme: OpenID Connect (OIDC)

> For authenticating with OpenID Connect, there is one method:

Reviewer: This and and list below it can be removed.

- [Protected Resource](#irods-as-an-oauth-protected-resource)

#### iRODS as an OAuth Protected Resource
While running in this mode, the HTTP API does not provide any grants to which you can authenticate.
Instead, you must implement an OAuth client yourself, authenticate, and provide an access token as a

> Bearer token when querying the HTTP API endpoints.

Reviewer: The section title and paragraph that follows can be combined. We don't need a title for protected resource mode since it's the only mode. You can mention protected resource mode in the paragraph.

Author: Done.
